File: 0x00-pascal_triangle/0-pascal_triangle.py
#!/usr/bin/python3
""" pascal triangle written in python
"""


def pascal_triangle(n):
    """ returns pascal triangle
    """
    if n <= 0:
        return []

    triangle = []

    for i in range(n):
        mylist = [1]
        """if its the first digit then use"""
        if i == 0:
            """mylist as the first digit"""
            triangle.append(mylist)
            continue
        else:
            """get the last row"""
            prev_row = triangle[-1]
            for j in range(1, i):
                nex = prev_row[j - 1] + prev_row[j]
                mylist.append(nex)
        """append 1 at the end of the list"""
        mylist.append(1)
        """the first digit should always be mylist"""
        triangle.append(mylist)

    return triangle

File: 0x00-pascal_triangle/0-pascal_triangle.cs
/** c-sharp solution **/
using System;
using System.Collections.Generic;

namespace PascalTriangle
{
    class Program
    {
        static List<List<int>> GeneratePascalTriangle(int n)
        {
            /**var triangle = List<List<int>>();**/
            List<List<int>> triangle = new List<List<int>>();

            if (n <= 0)
                return triangle;

            for (int i = 0; i < n; i++)
            {
                List<int> row = new List<int> { 1 };
                if (i == 0)
                {
                    triangle.Add(row);
                    continue;
                }
                else
                {
                    List<int> prevRow = triangle[triangle.Count - 1];
                    for (int j = 1; j < i; j++)
                    {
                        int nextValue = prevRow[j - 1] + prevRow[j];
                        row.Add(nextValue);
                    }
                }
                row.Add(1);
                triangle.Add(row);
            }
            return triangle;
        }
    }
}
